src/backend/marsha/core/api/video.py

```diff
@@ -723,6 +737,100 @@ def push_attendance(self, request, pk=None):
             status=404,
         )
 
+    @action(detail=False, methods=["post"])
+    # pylint: disable=unused-argument
+    def init_display_username(self, request, pk=None):
+        """View handling getting display_username. Create or get registration."""
+        serializer = serializers.LiveRegistrationDisplayUsernameSerializer(
+            data=request.data
+        )
+        if not serializer.is_valid():
+            return Response(
+                {"detail": "Invalid request."}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        if self.is_lti_token():
+            liveregistration, created = self.get_liveregistration_from_lti()
+            # liveregistration existed, we need to update username if necessary
+            if not created:
+                username = self.request.user.token.payload.get("user").get("username")
+                if username and liveregistration.username != username:
+                    liveregistration.username = username
+                    liveregistration.save()
+        else:
+            liveregistration, _ = LiveRegistration.objects.get_or_create(
+                anonymous_id=serializer.validated_data["anonymous_id"],
+                video=get_object_or_404(Video, pk=self.request.user.id),
+            )
+        return Response(
+            {
+                "display_username": liveregistration.display_username,
+                "username": liveregistration.username,
+            }
+        )
+
+    @action(detail=False, methods=["patch"], url_path="set_display_username")
+    # pylint: disable=unused-argument
+    def set_display_username(self, request, pk=None):
```

Comment:
Like I wrote on the `POST` endpoint. I think the url should be `/api/liveregistrations/display_username` and the resource should not be created if it doesn't exist. An error should be return. A `PATCH` is not able to create a resource.